From 7dbb331d99e8d4d4dc5b585f255b91ba4c8a20b9 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Sun, 18 Dec 2016 14:20:30 -0500
Subject: [PATCH] cmake: link against libmbedcrypto if available

In recent versions of mbed TLS, several symbols are moved to
libmbedcrypto.

Fixes: https://github.com/transmission/transmission/issues/115
---
 cmake/FindPolarSSL.cmake | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cmake/FindPolarSSL.cmake b/cmake/FindPolarSSL.cmake
index e4e1ac6..cd36c48 100644
--- a/cmake/FindPolarSSL.cmake
+++ b/cmake/FindPolarSSL.cmake
@@ -14,9 +14,14 @@ endif()
 
 find_path(MBEDTLS_INCLUDE_DIR NAMES mbedtls/version.h HINTS ${_MBEDTLS_INCLUDEDIR})
 find_library(MBEDTLS_LIBRARY NAMES mbedtls HINTS ${_MBEDTLS_LIBDIR})
+find_library(MBEDCRYPTO_LIBRARY NAMES mbedcrypto HINTS ${_MBEDTLS_LIBDIR})
 if(MBEDTLS_INCLUDE_DIR AND MBEDTLS_LIBRARY)
     set(POLARSSL_INCLUDE_DIR ${MBEDTLS_INCLUDE_DIR})
-    set(POLARSSL_LIBRARY ${MBEDTLS_LIBRARY})
+    if(MBEDCRYPTO_LIBRARY)
+        set(POLARSSL_LIBRARY ${MBEDTLS_LIBRARY} ${MBEDCRYPTO_LIBRARY})
+    else()
+        set(POLARSSL_LIBRARY ${MBEDTLS_LIBRARY})
+    endif()
     set(POLARSSL_VERSION ${_MBEDTLS_VERSION})
     set(POLARSSL_IS_MBEDTLS ON)
 else()
@@ -54,7 +59,7 @@ find_package_handle_standard_args(PolarSSL
         POLARSSL_VERSION
 )
 
-mark_as_advanced(MBEDTLS_INCLUDE_DIR MBEDTLS_LIBRARY POLARSSL_INCLUDE_DIR POLARSSL_LIBRARY)
+mark_as_advanced(MBEDTLS_INCLUDE_DIR MBEDTLS_LIBRARY MBEDCRYPTO_LIBRARY POLARSSL_INCLUDE_DIR POLARSSL_LIBRARY)
 
 if(POLARSSL_PREFER_STATIC_LIB)
     set(CMAKE_FIND_LIBRARY_SUFFIXES ${POLARSSL_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
-- 
2.7.4

